src/app.js

```diff
@@ -0,0 +1,21 @@
+import express from "express";
+import connectsToDb from "./config/dbConnect.js";
+import routes from "./routes/index.js";
+import errorHandler from "./middlewares/errorHandler.js";
+
+const app = express();
+routes(app);
+
+const connection = await connectsToDb();
+
+connection.on("error", (err) => {
+  console.error("Não foi possível se conectar ao banco de dados:", err);

```

Comment:
Uma observação: lembra que esse listener vai ser disparado em qualquer situação de erro de conexão com o MongoDB após a primeira conexão ter sido efetuada.

Para capturar o erro da primeira conexão com o MongoDB (ex.: url inválida), é preciso utilizar o catch do método `connect` do mongoose. Logo, no arquivo `dbConnect.js`, a chamada do método `connect` pode ser feita da seguinte forma:

```javascript
async function connectsToDB() {
  await mongoose.connect(process.env.DB_CONNECTION_STRING).catch((err) => {
    throw err.message;
  });
  return mongoose.connection;
}
```

E daí você pode manter esse listener para outras situações de erro de conexão do MongoDB (conexão caiu, ou houve alguma outra falha, etc). Daí pode por uma mensagem mais genérica, algo do tipo:

```suggestion
  console.error("an error occurred on MongoDB connection:", err);
```

Referência: https://mongoosejs.com/docs/connections.html#error-handling

Reply:
Bom ponto!